test/test_jit.py

```diff
@@ -5620,6 +5620,120 @@ def backward_hook(module, grad_input, grad_output):
+            return outputs + 30
+
+        n = Net()
+        n.register_forward_pre_hook(forward_pre_hook)
+        scripted_n = torch.jit.script(n)
+
+        m = Net()
+        m.register_forward_hook(forward_hook)
+        scripted_m = torch.jit.script(m)
+
+        t = torch.tensor([3, 4, 5])
+        self.assertEqual(n(t), scripted_n(t))
+        self.assertEqual(m(t), scripted_m(t))
+
+    def test_scripting_forward_hook(self):
+        class Net(nn.Module):
+            def __init__(self):
+                super(Net, self).__init__()
+
+            def forward(self, x):
+                return x + x
+
+        def forward_hook(self, inputs, outputs):
+            return outputs + 30
+
+        n = Net()
+        n.register_forward_hook(forward_hook)
+        scripted_n = torch.jit.script(n)
+
+        t = torch.tensor([3, 4, 5])
+        self.assertEqual(n(t), scripted_n(t))
+
+    def test_multiple_forward_hooks(self):
+        class Net(nn.Module):
+            def __init__(self):
+                super(Net, self).__init__()
+
+            def forward(self, x):
+                return x + x
+
+        def forward_hook1(self, inputs, outputs):
+            return outputs + 30
+
+        def forward_hook2(self, inputs, outputs):
+            return outputs * 4
+
+        n = Net()
+        n.register_forward_hook(forward_hook1)
+        n.register_forward_hook(forward_hook2)
+        scripted_n = torch.jit.script(n)
+
+        t = torch.tensor([3, 4, 5])
+        self.assertEqual(n(t), scripted_n(t))
+
+    def test_scripting_forward_pre_hook(self):
+        class Net(nn.Module):
+            def __init__(self):
+                super(Net, self).__init__()
+
+            def forward(self, x):
+                return x + x
+
+        def forward_pre_hook(self, inputs):
+            # type: (Tuple[Tensor,]) -> Tuple[Tensor,]
+            return (inputs[0] + 10,)
+
+        n = Net()
+        n.register_forward_pre_hook(forward_pre_hook)
+        scripted_n = torch.jit.script(n)
+
+        t = torch.tensor([3, 4, 5])
+        self.assertEqual(n(t), scripted_n(t))
+
+    def test_load_with_hooks(self):
+        class Net(nn.Module):
+            def __init__(self):
+                super(Net, self).__init__()
+
+            def forward(self, x):
+                return x + x
+
+        def forward_pre_hook(self, inputs):
+            # type: (Tuple[Tensor,]) -> Tuple[Tensor,]
+            return (inputs[0] + 10,)
+
+        def forward_hook(self, inputs, outputs):
+            return outputs + 30
+
+        n = Net()
+        n.register_forward_pre_hook(forward_pre_hook)
+        n.register_forward_hook(forward_hook)
+        scripted_n = torch.jit.script(n)
+
+        scripted_n.save("model1.gz")
+        n_loaded1 = torch.jit.load("model1.gz")
+        n_loaded1.save("model2.gz")
+        n_loaded2 = torch.jit.load("model2.gz")
+
+        t = torch.tensor([3, 4, 5])
+        self.assertEqual(n(t), n_loaded2(t))
```

Comment:
@driazati How about this one? I am not sure I can use `self.checkModule` here. Is there another alternative or is `assertEqual` fine here?

Reply:
This one's fine as is since `checkModule` won't help check the serialization